cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -145,7 +145,7 @@ handleSimpleNode :: forall blk. RunNode blk
                  -> NodeCLI
                  -> NodeConfiguration
                  -> IO ()
-handleSimpleNode p trace nodeTracers nCli nc = do
+handleSimpleNode p trace nodeTracers nCli nc = exceptionHandler $ do
```

Comment:
I'm not sure this is the best place to do it. Should it be higher-up in the call hierarchy (in `cardano-shell` even)?

What about the other executables part of this repository, should they do the same thing?

Reply:
Reasonable question. I'll defer to @Jimbo4350 and @ksaric.

Reply:
We already had this type of "global" exception handling in `cardano-shell` but it didn't really work well.
We had a case where we where swallowing exceptions that were raised when `Ctrl + C` was called and the conclusion was that this type of exception handling should be done at the level where it was happening.
For more info, see https://github.com/input-output-hk/cardano-shell/issues/144 and the PRs https://github.com/input-output-hk/cardano-shell/pull/145 and https://github.com/input-output-hk/cardano-shell/pull/147

Reply:
> We already had this type of "global" exception handling in `cardano-shell` but it didn't really work well.
> We had a case where we where swallowing exceptions that were raised when `Ctrl + C` was called and the conclusion was that this type of exception handling should be done at the level where it was happening.
> For more info, see [input-output-hk/cardano-shell#144](https://github.com/input-output-hk/cardano-shell/issues/144) and the PRs [input-output-hk/cardano-shell#145](https://github.com/input-output-hk/cardano-shell/pull/145) and [input-output-hk/cardano-shell#147](https://github.com/input-output-hk/cardano-shell/pull/147)

I see. What if we only catch synchronous exceptions? I have updated this PR to do that.